spark/src/main/scala/io/substrait/spark/expression/ToSparkExpression.scala

```diff
@@ -68,15 +82,69 @@ class ToSparkExpression(
     Literal(UTF8String.fromString(expr.value()), ToSubstraitType.convert(expr.getType))
   }
 
+  override def visit(expr: SExpression.BinaryLiteral): Literal = {
+    Literal(expr.value().toByteArray, ToSubstraitType.convert(expr.getType))
+  }
+
   override def visit(expr: SExpression.DecimalLiteral): Expression = {
     val value = expr.value.toByteArray
     val decimal = DecimalUtil.getBigDecimalFromBytes(value, expr.scale, 16)
     Literal(Decimal(decimal), ToSubstraitType.convert(expr.getType))
   }
+
   override def visit(expr: SExpression.DateLiteral): Expression = {
     Literal(expr.value(), ToSubstraitType.convert(expr.getType))
   }
 
+  override def visit(expr: SExpression.PrecisionTimestampLiteral): Literal = {
+    Literal(
+      Util.toMicroseconds(expr.value(), expr.precision()),
```

Comment:
Is it generally safe to convert timestamp values like this? If we're going from nanoseconds to microseconds, this will be lossy. Would it better for reject plans with non-microsecond precision and force the callers to make their plans compatible by using microseconds explicitly?

Reply:
Yeah, maybe not, I changed it in fd01c2f262534e10e71abf729eac306795df40be to be an assert instead. 

Though we may want to make it e.g. a user-choosable option later, or something, since e.g. Spark and DataFusion sometimes disagree on which precision to use (for IntervalDay, Spark uses micros and DF millis), and it'd be sad if the two would then be fundamentally incompatible. But for now we can just be strict.